### ISSUE TITLE
App fails deploy, I think because of latest Blacklight update

### ISSUE THREAD

**Issue by jrochkind** [sciencehistory/scihist_digicoll#1445]
```
remote:        Uglifier::Error: Unexpected token: punc (.)
remote:        --
remote:         18229   const nodes = Array.prototype.slice.call(elements);
remote:         18230   nodes.forEach(function (element) {
remote:         18231     element.addEventListener('click', function (e) {
remote:         18232       Blacklight.handleSearchContextMethod.call(e.currentTarget, e);
remote:         18233     });
remote:         18234   });
remote:         18235 };
remote:         18236
remote:            => Blacklight.csrfToken = () => document.querySelector('meta[name=csrf-token]')?.content;
```

We probably need to revert #1437. 

And then diagnose and report to Blacklight.

**Comment by jrochkind** [comment]
`./bin/rake assets:precompile` does NOT reproduce in dev, whichis annoying. 

`RAILS_ENV=production ./bin/rake assets` on development... doesn't actually work, it complains that it's missing some ScihistDigicoll::Env values that the app needs to boot, which is also annoying! (and my fault, that we've set this up in a way it's hard to run in production-environment on a dev computer, which we rarely need to do, but here's a time).  i can probably make it work by setting enough Env vars. 

Without being able to reproduce, still trying to revert #1437 and see if it deploys to staging okay...  and it did. So it's definitely the blacklight update.

**Comment by jrochkind** [comment]
So, to reproduce on a local machine, I had to jump through some hurdles, because of Reasons (some our own making, some Rails). 

1. Copy `./config/environments/production.rb` to `./config/environments/development.rb` (don't commit this!)
2. Run `RAILS_SERVE_STATIC_FILES=true ./bin/rake assets:precompile`

This reproduced. 

It's being caused because Blacklight added some ES6 javascript, and we used [uglifier](https://github.com/lautis/uglifier) to "minify" Javascript, which does not support ES6. 

I think Blacklight is going to do another release which avoids the ES6, it was a mistake. 

In the meantime, I confirmed that swapping `uglifier` for `terser`, as recommended in the uglifier README, seems to resolve things. 

(As we don't support IE anymore ourselves, there is probably no concern with delivering ES6 to browsers, it's fine. But Blacklight may not have meant to do so).
